.eslintrc.js

```diff
@@ -22,6 +22,45 @@ module.exports = {
                 'ghost/ghost-custom/max-api-complexity': 'error'
             }
         },
+        {
+            files: 'core/server/data/migrations/versions/**',
+            excludedFiles: [
+                'core/server/data/migrations/versions/1.*/*',
+                'core/server/data/migrations/versions/2.*/*',
+                'core/server/data/migrations/versions/3.*/*'
+            ],
+            rules: {
+                'ghost/filenames/match-regex': ['error', '^(?:\\d{2}|\\d{4}(?:-\\d{2}){4})(?:-[a-zA-Z]+)+$', true]
+            }
+        },
+        {
+            files: 'core/server/data/migrations/versions/**',
+            rules: {
+                'no-restricted-syntax': ['warn', {
+                    selector: 'ForStatement',
+                    message: 'For statements can perform badly in migrations'
+                }, {
+                    selector: 'ForOfStatement',
+                    message: 'For statements can perform badly in migrations'
+                }, {
+                    selector: 'ForInStatement',
+                    message: 'For statements can perform badly in migrations'
+                }, {
+                    selector: 'WhileStatement',
+                    message: 'While statements can perform badly in migrations'
+                }, {
+                    selector: 'CallExpression[callee.property.name=\'forEach\']',
+                    message: 'Loop constructs like forEach can perform badly in migrations'
+                }, {
+                    selector: 'CallExpression[callee.object.name=\'_\'][callee.property.name=\'each\']',
+                    message: 'Loop constructs like _.each can perform badly in migrations'
+                }, {
+                    selector: 'CallExpression[callee.property.name=/join|innerJoin|leftJoin/] CallExpression[callee.property.name=/join|innerJoin|leftJoin/] CallExpression[callee.name=\'knex\']',
+                    message: 'Use of multiple join statements in a single knex block'
+                }],
+                'ghost/no-return-in-loop/no-return-in-loop': ['error']
```

Comment:
Really surprised this doesn't exist in eslint, eslint-plugin-node or eslint-plugin-unicorn... seems like a doozie?!

Reply:
Yeah I couldn't find it! It's more involved than a `no-restricted-syntax` plugin because it needs to avoid traversing into function expressions. I've not enabled it globally but it sounds like a good one to introduce